### PR TITLE
CTCLoss dynamism support

### PR DESCRIPTION
**Pull request by chenhu-wang**
### Details:
 - *CTCLoss dynamism support*

### Tickets:
 - *70990*

### CODE REVIEW

File: inference-engine/tests/functional/plugin/cpu/single_layer_tests/ctc_Loss.cpp [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+ngraph::ParameterVector makeDynamicParams(const std::vector<ElementType>& types,
+                                          const std::vector<ov::PartialShape>& shapes) {
+    ngraph::ParameterVector outs;
+    NGRAPH_CHECK(types.size() == shapes.size());
+    for (size_t i = 0; i < types.size(); i++) {
+        auto paramNode = std::make_shared<ov::op::v0::Parameter>(types[i], shapes[i]);
+        outs.push_back(paramNode);
+    }
+    return outs;
+}
```
Comment (mandrono) [inline inference-engine/tests/functional/plugin/cpu/single_layer_tests/ctc_Loss.cpp:40]: better to place it inside `openvino/inference-engine/tests/ngraph_helpers/ngraph_functions/include/ngraph_functions/builders.hpp`

File: inference-engine/tests/functional/plugin/cpu/single_layer_tests/ctc_Loss.cpp [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+                                            bool,                           // preprocessCollapseRepeated
+                                            bool,                           // ctcMergeRepeated
+                                            bool,                           // unique
+                                            ElementType,                    // fp precision for logits
+                                            ElementType                     // int precision for label and length
+                                            >;
+ngraph::ParameterVector makeDynamicParams(const std::vector<ElementType>& types,
+                                          const std::vector<ov::PartialShape>& shapes) {
+    ngraph::ParameterVector outs;
+    NGRAPH_CHECK(types.size() == shapes.size());
```
Comment (mandrono) [inline inference-engine/tests/functional/plugin/cpu/single_layer_tests/ctc_Loss.cpp:34]:
```suggestion
    IE_ASSERT(types.size() == shapes.size());
```

File: inference-engine/tests/functional/plugin/cpu/single_layer_tests/ctc_Loss.cpp [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+        ov::PartialShape shapeN{inputDynamicShapesValues[0]};
+        ov::PartialShape shapeNT{inputDynamicShapesValues[0], inputDynamicShapesValues[1]};
+        ov::PartialShape shapeNTC{inputDynamicShapesValues[0], inputDynamicShapesValues[1], inputDynamicShapesValues[2]};
+        inputDynamicShapes = {shapeNTC, shapeN, shapeNT, shapeN};
+
+        std::vector<ElementType> types{fPrecision, iPrecision, iPrecision, iPrecision};
+        std::vector<ov::PartialShape> partialShapes{inputDynamicShapesValues, shapeN, shapeNT, shapeN};
+
+        auto params = makeDynamicParams(types, partialShapes);
+        auto bank = ngraph::op::Constant::create(ngraph::element::i64, ngraph::Shape{ }, {7});
```
Comment (mandrono) [inline inference-engine/tests/functional/plugin/cpu/single_layer_tests/ctc_Loss.cpp:106]: maybe better make it test parameter?

File: inference-engine/tests/functional/plugin/cpu/single_layer_tests/ctc_Loss.cpp [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+            },
+            // target
+            {
+                {{3, 6, 8}, {2, 5, 8}, {5, 6, 8}}
+            }
+        },
+        {
+            // dynamic
+            {
+                {-1, 10, 8},
```
Comment (mandrono) [inline inference-engine/tests/functional/plugin/cpu/single_layer_tests/ctc_Loss.cpp:223]: add instance with defined lower/upper bound please

File: inference-engine/tests/functional/plugin/cpu/single_layer_tests/ctc_Loss.cpp (hunk 1 above)
Comment (chenhu-wang) [reply inference-engine/tests/functional/plugin/cpu/single_layer_tests/ctc_Loss.cpp]: moved to nGraph helper. Thanks

File: inference-engine/tests/functional/plugin/cpu/single_layer_tests/ctc_Loss.cpp (hunk 2 above)
Comment (chenhu-wang) [reply inference-engine/tests/functional/plugin/cpu/single_layer_tests/ctc_Loss.cpp]: moved to nGraph helper. Thanks

File: inference-engine/tests/functional/plugin/cpu/single_layer_tests/ctc_Loss.cpp (hunk 3 above)
Comment (chenhu-wang) [reply inference-engine/tests/functional/plugin/cpu/single_layer_tests/ctc_Loss.cpp]: now the blank value is read from test parameters. Thanks!

File: inference-engine/tests/functional/plugin/cpu/single_layer_tests/ctc_Loss.cpp (hunk 4 above)
Comment (chenhu-wang) [reply inference-engine/tests/functional/plugin/cpu/single_layer_tests/ctc_Loss.cpp]: added, thanks.

File: inference-engine/tests/functional/plugin/cpu/single_layer_tests/ctc_Loss.cpp [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+        // ngraph::element::f16
+    };
+
+    const ngraph::element::TypeVector iPrecisions = {
+        ngraph::element::i32,
+        ngraph::element::i64
+    };
+
+    const std::vector<bool> preprocessCollapseRepeated = {true, false};
+    // TODO: [ticket:71088] there are some inconsistency between cpu plugin and ng ref
```
Comment (mandrono) [inline inference-engine/tests/functional/plugin/cpu/single_layer_tests/ctc_Loss.cpp:207]: is it correct issue number?
also better to add such test cases in openvino/inference-engine/tests/functional/plugin/cpu/shared_tests_instances/skip_tests_config.cpp

Comment (chenhu-wang) [reply inference-engine/tests/functional/plugin/cpu/single_layer_tests/ctc_Loss.cpp]: number corrected and all skip moved to skip_tests_config.cpp. thanks!

**Review by mandrono — APPROVED** [review]

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**